### PR TITLE
File System setup

### PR DESCRIPTION
**Pull request by satviksr**
Setup the basic FileSystem files

1. VBA File Mapper Mocker
1. Workbook open API with tests
1. Test Framework

### CODE REVIEW

File: FileSystem/FileMapperMocker.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+var USE_FILEMAPPER_MOCKER = true;
+
+/**
+ * Mocks the FileMapper module. To be replaced with complete implementation.
+ * The mocker has a list of hardcoded file path to google drive file id mappings.
+ *
+ * @return {Array.<string>} The selected text.
```
Comment (monikadhok) [inline FileSystem/FileMapperMocker.js]: What is this part? It's a declaration, what does return refer to?

File: FileSystem/TestFramework.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
+
```
Comment (monikadhok) [inline FileSystem/TestFramework.js]: Can you please add comments in this class?

File: FileSystem/Workbook.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+
+var FileSystem = {
+  current_directory: "<ToBeDefined>",
```
Comment (monikadhok) [inline FileSystem/Workbook.js:8]: Add a todo here to address current_directory work.

File: FileSystem/Workbook.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  api_workbook_open: api_workbook_open_,  
+};
+
+/**
+ * Emulates workbook.Open(filename) API 
+ * Opens the corresponding spreadsheet in a new new tab.
```
Comment (monikadhok) [inline FileSystem/Workbook.js]: new tab*

File: FileSystem/Workbook.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ *
+ * @param {string} path The local file path in the Excel file
+ * 
+ * @return {Spreadsheet} The spreadsheet object of the path
+ */
+function api_workbook_open_(path) {
+    var fileId = FileMapper.getFileId(this.current_directory, path, MimeType.GOOGLE_SHEETS);
+    var file = SpreadsheetApp.openById(fileId);
+    openURL(file.getUrl(), `"${file.getName()}" file open in new tab`);
+    return file;
```
Comment (monikadhok) [inline FileSystem/Workbook.js]: Do we need to return the file here? I guess openURL does it's job.

File: FileSystem/WorkbookTests.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
+/**
+ * For Testing only. Open spreadsheet in a new tab. 
+ *
+ * @param {string} path The local file path in the Excel file
+ * 
+ * @return {Spreadsheet} The spreadsheet object of the path
```
Comment (monikadhok) [inline FileSystem/WorkbookTests.js]: return?

File: FileSystem/WorkbookTests.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+ */
+function api_workbook_open(path) {
+  FileSystem.api_workbook_open(path);
+}
+
+/**
+ * Unit Tests
+ *
```
Comment (monikadhok) [inline FileSystem/WorkbookTests.js]: Find a linting tool, like JSLinter to auto format the code.

File: FileSystem/Workbook.js (hunk 5 above)
Comment (satviksr) [reply FileSystem/Workbook.js]: The return value of Workbook.open returns Workbook. So I thought it is best to replicate that behavior as well. Should I remove it? 

https://docs.microsoft.com/en-us/office/vba/api/excel.workbooks.open#return-value

File: FileSystem/Workbook.js (hunk 3 above)
Comment (satviksr) [reply FileSystem/Workbook.js:8]: Done

**Pull request closed without merging**